cluster-controller/src/main/java/io/strimzi/controller/cluster/Main.java

```diff
@@ -94,4 +106,38 @@ static CompositeFuture run(Vertx vertx, KubernetesClient client, Map<String, Str
         }
         return CompositeFuture.join(futures);
     }
+
+    static Future<Boolean> isOnOpenShift(Vertx vertx, KubernetesClient client)  {
+        URL kubernetesApi = client.getMasterUrl();
+        Future<Boolean> fut = Future.future();
+
+        HttpClientOptions httpClientOptions = new HttpClientOptions();
+        httpClientOptions.setDefaultHost(kubernetesApi.getHost());
+
+        if (kubernetesApi.getPort() == -1) {
+            httpClientOptions.setDefaultPort(kubernetesApi.getDefaultPort());
+        } else {
+            httpClientOptions.setDefaultPort(kubernetesApi.getPort());
+        }
+
+        if (kubernetesApi.getProtocol().equals("https")) {
+            httpClientOptions.setSsl(true);
+            httpClientOptions.setTrustAll(true);
+        }
+
+        HttpClient httpClient = vertx.createHttpClient(httpClientOptions);
+
+        httpClient.getNow("/oapi", res -> {
+            if (res.statusCode() == 200) {
```

Comment:
Just a nit ... using `HttpResponseStatus.OK.code()` instad of hardcoded `200`

Reply:
Fixed. I knew there was some class, but I didn't remember the name. Thanks for the comment.